beefy-gadget/src/worker.rs

```diff
@@ -212,27 +212,28 @@ where
 	}
 
 	fn handle_finality_notification(&mut self, notification: FinalityNotification<B>) {
-		debug!(target: "beefy", "🥩 Finality notification: {:?}", notification);
+		trace!(target: "beefy", "🥩 Finality notification: {:?}", notification);
 
 		// update best GRANDPA finalized block we have seen
 		self.best_grandpa_block = *notification.header.number();
 
 		if let Some(active) = self.validator_set(&notification.header) {
-			debug!(target: "beefy", "🥩 Active validator set id: {:?}", active);
-
-			metric_set!(self, beefy_validator_set_id, active.id);
-
-			// BEEFY should produce a signed commitment for each session
-			if (active.id != GENESIS_AUTHORITY_SET_ID) && (active.id != (1 + self.last_signed_id)) {
-				metric_inc!(self, beefy_skipped_sessions);
-			}
-
 			// Authority set change or genesis set id triggers new voting rounds
 			//
 			// TODO: (adoerr) Enacting a new authority set will also implicitly 'conclude'
 			// the currently active BEEFY voting round by starting a new one. This is
-			// temporary and needs to be repalced by proper round life cycle handling.
-			if (active.id != self.rounds.validator_set_id()) || (active.id == GENESIS_AUTHORITY_SET_ID) {
+			// temporary and needs to be replaced by proper round life cycle handling.
+			if active.id != self.rounds.validator_set_id()
+				|| (active.id == GENESIS_AUTHORITY_SET_ID && self.best_beefy_block.is_none())
```

Comment:
On genesis we only do this once instead of for all finalized blocks as we were doing previously.